src/main/java/aeneas/views/BuildLevelView.java

```diff
@@ -171,24 +168,25 @@ public void initialize(URL location, ResourceBundle resources) {
     // Handle changes in the level type.
     // TODO: Consider moving this to a separate class.
     levelType.selectedToggleProperty()
-        .addListener((ObservableValue<? extends Toggle> ov, Toggle toggle, Toggle new_toggle) -> {
-          if (new_toggle != null && !toggle.equals(new_toggle)) {
-            LevelWidgetView view = (LevelWidgetView) ((RadioButton) new_toggle).getUserData();
-            Level newLevel = view.resetLevelModel(this.model.getActiveLevel());
-            IMove move = new ChangeLevelTypeMove(this, newLevel);
-            if (move.execute()) model.addNewMove(move);
-            this.settingsBox.getChildren().set(1, view.getPanel());
-            this.levelView = view;
-          }
-        });
+    .addListener((ObservableValue<? extends Toggle> ov, Toggle toggle, Toggle new_toggle) -> {
+      if (new_toggle != null && !toggle.equals(new_toggle)) {
+        LevelWidgetView view = (LevelWidgetView) ((RadioButton) new_toggle).getUserData();
+        Level newLevel = view.resetLevelModel(this.model.getActiveLevel());
+        IMove move = new ChangeLevelTypeMove(this.model, newLevel);
+        if (move.execute()) model.getActiveLevel().addNewMove(move);
+        this.settingsBox.getChildren().set(1, view.getPanel());
+        this.levelView = view;
+        this.undoController.setLevel(model.getActiveLevel());
```

Comment:
Ew. Make the UndoController take the model and then have it call `getActiveLevel` every time it needs the level. Since we swap out the level at weird times, no one other than the Model should maintain a reference to the active level.